infra/Pos/Subscription/Common.hs

```diff
@@ -60,17 +65,34 @@ data SubscriptionTerminationReason =
 -- giving the reason. Notices will be logged before and after the subscription.
 subscribeTo
     :: forall m. (SubscriptionMode m)
-    => SendActions m -> NodeId -> m SubscriptionTerminationReason
-subscribeTo sendActions peer = do
+    => Timer -> SendActions m -> NodeId -> m SubscriptionTerminationReason
+subscribeTo keepAliveTimer sendActions peer = do
     logNotice $ msgSubscribingTo peer
-    outcome <- try $ withConnectionTo sendActions peer $ \_peerData ->
-           pure $ Conversation $ \conv -> do
-               send conv MsgSubscribe
-               recv conv 0 :: m (Maybe Void) -- Other side will never send
+    outcome <- try $ withConnectionTo sendActions peer $ \_peerData -> NE.fromList
+        -- Sort conversations in descending order based on their version so that
+        -- the highest available version of the conversation is picked.
+        [ Conversation convMsgSubscribe
+        , Conversation convMsgSubscribe1
+        ]
```

Comment:
Ah and the order matters here too doesn't it?

Reply:
According to my experiments this is the only place where the order matters.

Reply:
Yes, order here matters: first matched is to be picked up